db/migrate/20150612135546_create_host_status.rb

```diff
@@ -33,4 +40,41 @@ def down
 
     drop_table :host_status
   end
+
+  private
+
+  def update_statuses(host)
+    results = HostStatus.status_registry.map do |status_class|
+      status = host.get_status(status_class)
+      update_sub_status(status)
+    end
+
+    if results.any?
+      host.host_statuses.reload
+      host.refresh_global_status
+
+      unless (saved = host.save(:validate => false))
+        logger.warn "Skipping global status refresh for host #{host.name}"
+      end
+      results.push saved
+    end
+
+    !results.include?(false)
```

Comment:
do we care about which one failed? is there a follow up action? if there is none, I'm not sure if it helps the user know that something failed be he has no idea what and if he should do something about it.